### ISSUE TITLE
Duplicate close tab confirmation

### ISSUE THREAD

**Issue by hklene** [piroor/treestyletab#2484]
## Short description
Closing inner tree node with folded children triggers multiple confirmations

## Steps to reproduce
produce Tree like this:

- A parent node to keep e.g. search result
  - B one entry to fold & close
    - b1 detail, e.g. full-scale detail pictures from B
    - b2 detail
    - b3 detail
    - b4 detail

go to Tab B
fold the children, so they become invisible
close Tab B
confirm to close 5 Tabs total

## Expected result
Tabs B and below are gone

## Actual result
secondary confirm to close another 2 Tabs pops up
cancel it, will leave some random tabs behind

## Speculation
On close of an inner tree-tab, the algorithm tries to replace it by it's first child.
This child (e.g. b3) now becomes the new parent of the remaining tabs (b4)
But there is a close operation pending for b3, that will trigger the secondary confirm.

Closing multiple Tabs would have to prune only leaves of the tree in a depth first walk. These would never have a remaining child, so the dialog would not trigger again. Detecting the situation and suppressing the dialog might open a can of worms of slipping over conditions or even worse to not handle a concurrent close operation triggered separately by a second user interaction.

The steps to reproduce seem to be timing dependent. It might help to add more details tabs b5-b... and to add more parent level to B, so the way to the root becomes longer, as well as adding more complex siblings to A or even more windows with open tabs? Contrary moving B to be the root itself or adding siblings to B might suppress the error - perhaps there is already incomplete detection in place #1851 ?

## Workaround
Keep confirming until the tabs are gone or unfold the tree and close tabs one by one

## Environment

 * Platform (OS): Kubuntu 19.04
 * Version of Firefox: 72.0.1
 * Version (or revision) of Tree Style Tab: 3.3.5 (2020.1.24):

**Comment by piroor** [comment]
I agree TST possibly has a duplicated confirmation problem, but sadly I couldn't reproduce this problem on my environment. Steps I did are:

1. Start Firefox with clean profile.
2. Load TST (I've used the latest development build.)
3. Press the Ctrl-L shortcut to focus to the location bar.
4. Input "foxkeh" and hit the Enter key to search it on the Web (by Google).
5. Middle-click the first search result ( http://foxkeh.jp/ for me) and switch to the tab by clicking on the tab in the sidebar.
6. Middle-click on links in the page 4 times.
7. Collapse the tree. Then finally I got a tree like:
   - A: search result tab
     - B: foxkeh.jp (active tab, with collapsed subtree)
       - C: foxkeh.jp (collapsed)
       - D: foxkeh.jp (collapsed)
       - E: foxkeh.jp (collapsed)
       - F: foxkeh.jp (collapsed)
8. Press the Ctrl-W keyboard shortcut to close the active tab.
9. The tab B is closed by Firefox, and a confirmation dialog for closing 5 tabs is shown in the sidebar.
10. Click "Close tabs" or "Cancel".
    - When I click the "Close tabs" button, rest 4 tabs (C-F) are also closed.
    - When I click the "Cancel" button, the tab B is restored.

Any point different to your case?

**Comment by hklene** [comment]
Just today it happened again in my productive profile with > 250 tabs distributed over 4 windows and quite a few other add-ons ... But so far no luck reproducing in an empty profile ... I keep trying ...

**Comment by irvinm** [comment]
@hklene any updates?

**Comment by irvinm** [comment]
@piroor this seems to be inactive and we can close.
